### PR TITLE
Fix Optimization Summary failing after loading from hdf5

### PR DESCRIPTION
**Pull request by PaulJonasJost**
if free_indices were interpreted as floats, we could not use them to subset the array. Happened to me after loading results from hdf5

### CODE REVIEW

**Comment by dweindl** [comment]
> if free_indices were interpreted as floats, we could not use them to subset the array. Happened to me after loading results from hdf5

Thanks. However, I'd prefer fixing the root of the issue, rather than the symptom. I.e., ensuring `free_indices` is always `int`.

**Comment by PaulJonasJost** [comment]
> > if free_indices were interpreted as floats, we could not use them to subset the array. Happened to me after loading results from hdf5
> 
> Thanks. However, I'd prefer fixing the root of the issue, rather than the symptom. I.e., ensuring `free_indices` is always `int`.

So the [save function](https://github.com/ICB-DCM/pyPESTO/blob/872e2530e55c944ee55f0e38bdbdb479d0c7fba5/pypesto/store/hdf5.py#L24) should have taken care of this. Also `Problem` defines their free indices as `int`, therefore I am not sure where it went wrong except for the integration of [free indices to the Optimizer result](https://github.com/ICB-DCM/pyPESTO/blob/872e2530e55c944ee55f0e38bdbdb479d0c7fba5/pypesto/result/optimize.py#L198) guaranteed it here, but will have one more look whether anything else might be missing. Will still leave it in the summary as well as a second safety measure.

**Comment by dweindl** [comment]
> not sure where it went wrong

I don't see either how this could become `float`, except for the case of empty `x_free_indices`, which would be fixed by adding `dtype` to https://github.com/ICB-DCM/pyPESTO/blob/872e2530e55c944ee55f0e38bdbdb479d0c7fba5/pypesto/result/optimize.py#L198.

**Comment by PaulJonasJost** [comment]
@dweindl could you re-review please :) added the `dtype` argument but still kept the change before as an additional safety measure.

**Review by dweindl — COMMENTED** [review]
> still kept the change before as an additional safety measure

I don't think it makes sense to have it there. Either we specify and ensure that `problem.x_free_indices` contains `int` and assume that everywhere, or we specify it to be of any type of number and treat it accordingly everywhere.

File: pypesto/result/optimize.py [hunk 1]
```diff
@@ -152,27 +152,30 @@ def summary(self, full: bool = False, show_hess: bool = True) -> str:
...
+            free_indices = self.free_indices.astype(int)
         message = (
             "### Optimizer Result\n\n"
             f"* optimizer used: {self.optimizer}\n"
             f"* message: {self.message} \n"
             f"* number of evaluations: {self.n_fval}\n"
             f"* time taken to optimize: {self.time:0.3f}s\n"
-            f"* startpoint: {self.x0 if full or self.x0 is None else self.x0[self.free_indices]}\n"
-            f"* endpoint: {self.x if full else self.x[self.free_indices]}\n"
+            f"* startpoint: "
```
[the remark this answers is not in the document]
Comment (PaulJonasJost) [reply pypesto/result/optimize.py:163]: is that actually something one should do? I always see these strings as one big string and therefore they should either all have an f"" or none 🤔

[the remark this answers is not in the document]
Comment (dweindl) [reply pypesto/result/optimize.py:163]: In my opinion, if a string doesn't need an `f`, it shouldn't have one. The only (weak) reason I could think of for using it here would be source code alignment, but it isn't even done consistently.

See also https://docs.astral.sh/ruff/rules/f-string-missing-placeholders/

**Comment by PaulJonasJost** [comment]
> > still kept the change before as an additional safety measure
> 
> I don't think it makes sense to have it there. Either we specify and ensure that `problem.x_free_indices` contains `int` and assume that everywhere, or we specify it to be of any type of number and treat it accordingly everywhere.

IMO, it does make sense to do this here, as we only **now** start to ensure it is int, but for backwards compatibility or previous results, it might be good to have this as an additional safety net?

**Comment by PaulJonasJost** [comment]
Closing this for now.

**Pull request closed without merging**